mLRS/Common/hal/esp-rxclock.h

```diff
@@ -119,47 +70,30 @@ void RxClockBase::Init(uint16_t period_ms)
     CLOCK_PERIOD_10US = period_ms * 100; // frame rate in units of 10us
     doPostReceive = false;
 
-    init_isr_off();
-    enable_isr();
+    // Initialise the timer
+    timer1_attachInterrupt(CLOCK_IRQHandler); 
+    timer1_enable(TIM_DIV16, TIM_EDGE, TIM_LOOP);
+    timer1_write(50); //5MHz (5 ticks/us - 1677721.4 us max), 50 ticks = 10us
+    Reset();
 }
 
+void RxClockBase::disable_isr(void)
+{
+    timer1_detachInterrupt(); 
+}
 
 void RxClockBase::SetPeriod(uint16_t period_ms)
 {
     CLOCK_PERIOD_10US = period_ms * 100;
 }
 
-
 void RxClockBase::Reset(void)
 {
     if (!CLOCK_PERIOD_10US) while (1) {}
 
-    __disable_irq();
-    uint32_t CNT = espTIMER.CNT; // works for both 16 and 32 bit timer
-    espTIMER.CCR1 = CNT + CLOCK_PERIOD_10US;
-    espTIMER.CCR3 = CNT + CLOCK_SHIFT_10US;
-    espTIMER.CC1_FLAG = false; // important to do
-    espTIMER.CC3_FLAG = false;
-    __enable_irq();
-}
-
-
-void RxClockBase::init_isr_off(void)
-{
-    ITimer.attachInterruptInterval(10, CLOCK_IRQHandler);
-    ITimer.disableTimer();
-}
-
-
-void RxClockBase::enable_isr(void)
-{
-    ITimer.enableTimer();
-}
-
-
-uint16_t RxClockBase::tim_10us(void)
-{
-    return espTIMER.CNT; // return 16 bit even for 32 bit timer
+    CCR1 = CNT_10us + CLOCK_PERIOD_10US;
```

Comment:
we need to find something to protect that block, replacing the disable_irq(), enable_irq() which were enclosing the block
if there isn't a proper global isr enable/disable, the RTOS shoudl provide some function to protect critical blocks